llm/server.go

```diff
@@ -864,6 +865,52 @@ func (s *llmServer) Embedding(ctx context.Context, prompt string) ([]float64, er
 		return nil, fmt.Errorf("unexpected server status: %s", status.ToString())
 	}
 
+	switch prompts := prompts.(type) {
+	case string:
+		// single prompt
+		embedding, err := s.EmbeddingSingle(ctx, prompts)
+		if err != nil {
+			return nil, err
+		}
+		return [][]float64{embedding}, nil
+	case []string:
+		// multiple prompts
+		errCh := make(chan error, 1)
+		successCh := make(chan [][]float64, 1)
+		num_prompts := len(prompts)
+		embeddings := make([][]float64, num_prompts)
+		var wg sync.WaitGroup
+		wg.Add(num_prompts)
+		for i, p := range prompts {
+			go func(i int, p string) {
+				defer wg.Done()
+				slog.Info("embedding", "prompt", p)
+				embedding, err := s.EmbeddingSingle(ctx, p)
+				if err != nil {
+					errCh <- err
+					return
+				}
+				embeddings[i] = embedding
+			}(i, p)
+		}
+
+		go func() {
+			wg.Wait()
+			successCh <- embeddings
+		}()
+
+		select {
+		case err := <-errCh:
+			return nil, err
+		case embeddings := <-successCh:
+			return embeddings, nil
+		}
+	default:
+		return nil, fmt.Errorf("unsupported prompt type: %T", prompts)
+	}
+}
+
+func (s *llmServer) EmbeddingSingle(ctx context.Context, prompt string) ([]float64, error) {
```

Comment:
we shouldn't need this, llmServer should just accept `[]string` and the API router/frontend can deal with creating a `[]string{prompt}` if `prompt` is a string